plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -78,10 +82,22 @@
     private final S3BlobStore blobStore;
     private final String keyPath;
 
+    private final RequestMetricCollector listMetricCollector;
+
     S3BlobContainer(BlobPath path, S3BlobStore blobStore) {
         super(path);
         this.blobStore = blobStore;
         this.keyPath = path.buildAsString();
+        this.listMetricCollector = new RequestMetricCollector() {
```

Comment:
You could just keep these collectors as fields on `S3BlobStore` and have getters for them on it?
Then you save adding new fields to the retrying input stream and containers and don't have to create a new instance for every container + get request.